repos/system_upgrade/el7toel8/actors/selinux/selinuxcontentscanner/tests/component_test.py

```diff
@@ -0,0 +1,114 @@
+import os
+
+import pytest
+
+from leapp.snactor.fixture import current_actor_context
+from leapp.models import SELinuxModule, SELinuxModules, SELinuxCustom, SELinuxFacts, SELinuxRequestRPMs
+from leapp.libraries.stdlib import api, run, CalledProcessError
+from leapp.reporting import Report
+
+test_modules = [
+    ["400", "mock1"],
+    ["99", "mock1"],
+    ["200", "mock1"],
+    ["400", "mock2"],
+    ["999", "mock3"]
+]
+
+semanage_commands = [
+    ['fcontext', '-t', 'httpd_sys_content_t', '"/web(/.*)?"'],
+    ['fcontext', '-t', 'ganesha_var_run_t', '"/ganesha(/.*)?"'],
+    ['port', '-t', 'http_port_t', '-p', 'udp', '81'],
+    ['permissive', 'abrt_t']
+]
+
+testmoduledir = os.path.join(os.getcwd(), "tests/mock_modules/")
+
+
+# Test disabled because it's setup and teardown would modify the system
+# Remove "_" before re-activation
+def setup_():
+    for priority, module in test_modules:
+        try:
+            run(["semodule", "-X", priority, "-i", os.path.join(testmoduledir, module + ".cil")])
+        except CalledProcessError as e:
+            api.current_logger().warning("Error installing mock module: %s", str(e.stderr))
+            continue
+
+    for command in semanage_commands:
+        try:
+            run(["semanage", command[0], "-a"] + command[1:])
+        except CalledProcessError as e:
+            api.current_logger().warning("Error applying selinux customizations %s", str(e.stderr))
+            continue
+
+
+def findModule(selinuxmodules, name, priority):
+    for module in selinuxmodules.modules:
+        if module.name == name and module.priority == int(priority):
+            return module
+    return None
+
+
+def findSemanageRule(rules, rule):
```

Comment:
Code duplication?

Reply:
Yes. I was told it's better than trying to set up sharing code between tests because of 1 function.